### PR TITLE
[K32W] Add NXP platform in the Github workflows

### PR DESCRIPTION
**Pull request by doru91**
This adds K32W061 in the Github workflows.

Basically it continues the work from https://github.com/project-chip/connectedhomeip/pull/5430 - compared to that PR it just updates the Docker file for installing some new packages and increments the version number -as suggested by @andreilitvin

### CODE REVIEW

File: integrations/docker/images/chip-build/version [hunk 1]
```diff
@@ -1 +1 @@
-0.4.21
```
Comment (Damian-Nordic) [inline integrations/docker/images/chip-build/version:1]: This should only be incremented when a Dockerfile is changed. Is it needed?

File: integrations/docker/images/chip-build-k32w/Dockerfile [hunk 2]
```diff
@@ -5,6 +5,7 @@ FROM connectedhomeip/chip-build:${VERSION}
 RUN set -x \
     && apt-get update \
     && DEBIAN_FRONTEND=noninteractive apt-get install -y wget unzip \
+    && pip3 install pycrypto && pip3 install pycryptodome \
```
[the remark this answers is not in the document]
Comment (doru91) [reply integrations/docker/images/chip-build-k32w/Dockerfile:8]: Should I create another PR only with the above change (pip3 install pycrypto && pip3 install pycryptodome)? Then I believe I need to talk to Andrei Litvin to update the Docker image on the repo because I don't have push access. Is this the correct approach?

[the remark this answers is not in the document]
Comment (Damian-Nordic) [reply integrations/docker/images/chip-build-k32w/Dockerfile]: I mean, we can't add workflows which break CI, so if the workflow requires Docker changes it should be added after the Docker image has been updated. If you can skip signing step for now so that the workflow completes without an error it's probably the fastest way to have your platform verified in CI.

File: integrations/docker/images/chip-build/version (hunk 1 above)
Comment (andy31415) [inline integrations/docker/images/chip-build/version:1]: why this change?

File: .github/workflows/examples-k32w.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    pull_request:
+
+jobs:
+    k32w:
+        name: K32W
+        env:
+            BUILD_TYPE: gn_k32w
+
+        runs-on: ubuntu-latest
+        if: github.actor != 'restyled-io'
```
[the remark this answers is not in the document]
Comment (doru91) [reply .github/workflows/examples-k32w.yaml]: Thanks, also updated in this PR.

File: .github/workflows/examples-k32w.yaml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                - "/tmp/bloat_reports:/tmp/bloat_reports"
+                - "/tmp/output_binaries:/tmp/output_binaries"
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v2
+              # Fetch depth 0 to get all history and be able to check mergepoint for bloat report
+              with:
+                  fetch-depth: 0
+                  submodules: true
+            - name: Initialize CodeQL
```
[the remark this answers is not in the document]
Comment (doru91) [reply .github/workflows/examples-k32w.yaml]: Good, I also commented for this PR.

File: scripts/examples/k32w_example.sh [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+source "$(dirname "$0")/../../scripts/activate.sh"
+
+set -x
+env
+
+./third_party/k32w_sdk/mr2_fixes/patch_k32w_mr2_sdk.sh
+gn gen --root="$1" "$2" --args="k32w_sdk_root=\"${K32W061_SDK_ROOT}\" is_debug=true"
+ninja -C "$2"
+$K32W061_SDK_ROOT/tools/imagetool/sign_images.sh "$2"
```
[the remark this answers is not in the document]
Comment (doru91) [reply scripts/examples/k32w_example.sh]: Skipped it for the moment.

File: integrations/docker/images/chip-build/version (hunk 1 above)
Comment (doru91) [reply integrations/docker/images/chip-build/version]: Not needed anymore since I decided to avoid modifying the Docker file.

**Review by Damian-Nordic — APPROVED** [review]

**Comment by doru91** [comment]
Is there any problem with "restyled"? I tried to rebase several times to restart the "restyled" process but it keeps failing.

**Comment by bzbarsky-apple** [comment]
@doru91 From the log:

> stderr ! [remote rejected]   restyled/pull-5724 -> restyled/pull-5724 (refusing to allow a GitHub App to create or update workflow `.github/workflows/examples-k32w.yaml` without `workflows` permission)
> stdoutSetting status of error for 8490abf
> stderrerror: failed to push some refs to 'https<SCRUBBED>@github.com/project-chip/connectedhomeip.git'

So sounds like it ran into style problems in `examples-k32w.yaml` but can't create the normal restyle PR because it doesn't have permissions to change that file?

Running restyled locally would probably help....

**Review by msandstedt — APPROVED** [review]

**Comment by doru91** [comment]
> @doru91 From the log:
> 
> > stderr ! [remote rejected]   restyled/pull-5724 -> restyled/pull-5724 (refusing to allow a GitHub App to create or update workflow `.github/workflows/examples-k32w.yaml` without `workflows` permission)
> > stdoutSetting status of error for [8490abf](https://github.com/project-chip/connectedhomeip/commit/8490abf9a06b16577a04c9f28af35feafb8e3852)
> > stderrerror: failed to push some refs to 'https@github.com/project-chip/connectedhomeip.git'
> 
> So sounds like it ran into style problems in `examples-k32w.yaml` but can't create the normal restyle PR because it doesn't have permissions to change that file?
> 
> Running restyled locally would probably help....

Solved, I think it can be merged.

**Review by jmartinez-silabs — APPROVED** [review]
